File: pinyougou-sellergoods-interface/src/main/java/com/pinyougou/sellergoods/service/BrandService.java
package com.pinyougou.sellergoods.service;

import java.util.List;
import java.util.Map;

import com.pinyougou.pojo.TbBrand;
import entity.PageResult;

/**
 * 品牌接口
 *
 * @author Administrator
 */
public interface BrandService {

    //查询品牌列表
    public List<TbBrand> findAll();

    //分页查询品牌列表
    public PageResult findPage(Integer pageNum, Integer pageSize);

    /**
     * 品牌管理:增加
     */
    public void add(TbBrand brand);


    //修改
    void update(TbBrand brand);

    //根据id查找,用于回显
    TbBrand findOne(Long id);

    //多条删除
    void delete(Long[] ids);

    /**
     * *
     * @param pageNum 当前页 码
     * @param  pageSize 每页记录数
     */
    public PageResult findPage(TbBrand brand, int pageNum,int pageSize);

    /**
     * 品牌下拉框数据
     */
    List<Map> selectOptionList();
}

File: pinyougou-manager-web/src/main/webapp/js/service/brandService.js
app.service('brandService', function ($http) {
    //根据id查询Brand
    this.findOne = function (id) {
        return $http.get('../brand/findOne.do?id=' + id);
    };

    //新增
    this.add = function (entity) {
        return $http.post('../brand/add.do', entity);
    };
    //修改
    this.update = function (entity) {
        return $http.post('../brand/update.do', entity);
    };

    //删除
    this.dele = function (ids) {
        return $http.get('../brand/delete.do?ids=' + ids);
    };

    //条件查询

    this.search = function (page, rows, searchEntity) {
        return $http.post('../brand/search.do?page=' + page + '&rows=' + rows, searchEntity);
    }

    //关联品牌
    this.selectOptionList=function () {
       return $http.get('../brand/selectOptionList.do')
    }
});